File: backend/app/schemas/medicine.py
from datetime import date

from pydantic import BaseModel, ConfigDict


class MedicineCreate(BaseModel):
    name: str
    manufacturer: str
    price: float
    stock: int
    expiry_date: date
    prescription_required: bool = False

class MedicineUpdate(BaseModel):
    name: str
    manufacturer: str
    price: float
    stock: int
    expiry_date: date
    prescription_required: bool

class MedicineResponse(BaseModel):
    id: int
    name: str
    manufacturer: str
    price: float
    stock: int
    expiry_date: date
    prescription_required: bool

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/medicine.ts
export type Medicine = {
  id: number;
  name: string;
  manufacturer: string;
  price: number;
  stock: number;
  expiry_date: string;
  prescription_required: boolean;
};

export type MedicineCreate = Omit<Medicine, "id">;
export type MedicineUpdate = Partial<MedicineCreate>;